Ubuntu Disco (19.04) fails with node issues

#### Expected Behavior

A complete installation without issue.

-----

#### Actual Behavior

Build process fails, apparently due to node based on the log. 	

	## Confirming "disco" is supported...

	+ curl -sLf -o /dev/null 'https://deb.nodesource.com/node_8.x/dists/disco/Release'

	## Adding the NodeSource signing key to your keyring...

	+ curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
	OK

	## Creating apt sources list file for the NodeSource Node.js 8.x LTS Carbon repo...

	+ echo 'deb https://deb.nodesource.com/node_8.x disco main' > /etc/apt/sources.list.d/nodesource.list
	+ echo 'deb-src https://deb.nodesource.com/node_8.x disco main' >> /etc/apt/sources.list.d/nodesource.list

	## Running `apt-get update` for you...

	+ apt-get update
 
Where as on 18.04

	## Confirming "bionic" is supported...

	+ curl -sLf -o /dev/null 'https://deb.nodesource.com/node_8.x/dists/bionic/Release'

	## Adding the NodeSource signing key to your keyring...

	+ curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
	OK

	## Creating apt sources list file for the NodeSource Node.js 8.x LTS Carbon repo...

	+ echo 'deb https://deb.nodesource.com/node_8.x bionic main' > /etc/apt/sources.list.d/nodesource.list
	+ echo 'deb-src https://deb.nodesource.com/node_8.x bionic main' >> /etc/apt/sources.list.d/nodesource.list

	## Running `apt-get update` for you...

	+ apt-get update

With the correct version of node manually installed the service continues to reboot over and over again. 

	May 03 12:58:38 xoce xo-server[16303]: 2019-05-03T12:58:38.793Z - xo:xo-server - [ERROR] uncaught exception
	May 03 12:58:38 xoce xo-server[16303]: { error:
	May 03 12:58:38 xoce xo-server[16303]:    { Error: Could not locate the bindings file. Tried:
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/build/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/build/Debug/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/build/Release/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/out/Debug/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/Debug/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/out/Release/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/Release/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/build/default/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:     → /opt/xen-orchestra/packages/xo-server/compiled/10.15.2/linux/x64/leveldown.node
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.require (internal/modules/cjs/loader.js:637:17)
	May 03 12:58:38 xoce xo-server[16303]:        at require (internal/modules/cjs/helpers.js:22:18)
	May 03 12:58:38 xoce xo-server[16303]:        at Module._compile (internal/modules/cjs/loader.js:689:30)
	May 03 12:58:38 xoce xo-server[16303]:        at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
	May 03 12:58:38 xoce xo-server[16303]:        at Module.load (internal/modules/cjs/loader.js:599:32)
	May 03 12:58:38 xoce xo-server[16303]:        at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module._load (internal/modules/cjs/loader.js:530:3)
	May 03 12:58:38 xoce xo-server[16303]:        at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
	May 03 12:58:38 xoce xo-server[16303]:        at startup (internal/bootstrap/node.js:283:19)
	May 03 12:58:38 xoce xo-server[16303]:        at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
	May 03 12:58:38 xoce xo-server[16303]:      tries:
	May 03 12:58:38 xoce xo-server[16303]:       [ '/opt/xen-orchestra/packages/xo-server/build/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/build/Debug/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/build/Release/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/out/Debug/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/Debug/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/out/Release/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/Release/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/build/default/leveldown.node',
	May 03 12:58:38 xoce xo-server[16303]:         '/opt/xen-orchestra/packages/xo-server/compiled/10.15.2/linux/x64/leveldown.node' ] } }
	May 03 12:58:38 xoce systemd[1]: xo-server.service: Succeeded.
	May 03 12:58:39 xoce systemd[1]: xo-server.service: Service RestartSec=100ms expired, scheduling restart.
	May 03 12:58:39 xoce systemd[1]: xo-server.service: Scheduled restart job, restart counter is at 317.
	May 03 12:58:39 xoce systemd[1]: Stopped XO Server.
	May 03 12:58:39 xoce systemd[1]: Started XO Server.

--- a/xo_install.sh
+++ b/xo_install.sh
@@ -14,7 +14,6 @@ xo_branch="master"
 xo_server="https://github.com/vatesfr/xen-orchestra"
 n_repo="https://raw.githubusercontent.com/visionmedia/n/master/bin/n"
 yarn_repo="deb https://dl.yarnpkg.com/debian/ stable main"
-node_source="https://deb.nodesource.com/setup_8.x"
 yarn_gpg="https://dl.yarnpkg.com/debian/pubkey.gpg"
 n_location="/usr/local/bin/n"
 xo_server_dir="/opt/xen-orchestra"
@@ -25,18 +24,21 @@ xo_service="xo-server.service"
 /usr/bin/apt-get update
 /usr/bin/apt-get --yes install git curl
 
-#Install node and yarn
+#Install yarn
 cd /opt
 
-/usr/bin/curl -sL $node_source | bash -
+                                       
 /usr/bin/curl -sS $yarn_gpg | apt-key add -
 echo "$yarn_repo" | tee /etc/apt/sources.list.d/yarn.list
 /usr/bin/apt-get update
-/usr/bin/apt-get install --yes nodejs yarn
+/usr/bin/apt-get install --yes yarn
 
 # Install n
 /usr/bin/curl -o $n_location $n_repo
 /bin/chmod +x $n_location
+
+# Install node via n
+n 8.16
 
 # Symlink node directories
 ln -s /usr/bin/node /usr/local/bin/node

